### PR TITLE
Windows Subsystem for Linux build instructions

### PR DESCRIPTION
**Pull request by AnHeuermann**
Played around with WSL and wrote steps to reproduce successful build down.

Do you think this has a benefit for other people as well?

### CODE REVIEW

**Comment by AnHeuermann** [comment]
For the tests some additional programs are needed. Do we want to add them to build-dependencies of OpenModelica? They are not really needed for building, only for building all test suite dependencies and running the tests.

**Comment by AnHeuermann** [comment]
And I have still some failing tests with optimization and measuring time:
```     ./openmodelica/cppruntime/hpcom/Modelica.Electrical.Analog.Examples.CauerLowPassSC_level_omp_measureTime.mos
./openmodelica/cruntime/optimization/basic/TFC6.mos
./openmodelica/cruntime/optimization/basic/testAlgLoop6.mos
./openmodelica/cruntime/optimization/basic/testAlgLoop7.mos
./openmodelica/cruntime/optimization/basic/testDerInput.mos
./openmodelica/cruntime/optimization/benchmark/runDrumBoiler.mos
./simulation/modelica/built_in_functions/MeasureTime.mos
```
@JKRT Did you run the test suite on your WSL and got a better result? Probably I'm missing one or two programs still.

File: OMCompiler/README-Windows-WSL.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Install build dependencies
+See the Dependencies for Linux in [Linux instructions](README.Linux.md) or use
+```bash
+echo deb http://build.openmodelica.org/apt bionic nightly | sudo tee -a /etc/apt/sources.list
+echo deb-src http://build.openmodelica.org/apt bionic nightly | sudo tee -a /etc/apt/sources.list
+
+# Get key for OpenModelica
+# apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 0x3a59b53664970947   # Since there is a bug in WSL with $ apt-key adv
+curl -sL "http://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3A59B53664970947" | sudo apt-key add
```
Comment (AnHeuermann) [inline OMCompiler/README-Windows-WSL.md]: Is there a better way to get the key for OpenModelica for `apt-get update`?

Comment (sjoelund) [reply OMCompiler/README-Windows-WSL.md]: `wget -q http://build.openmodelica.org/apt/openmodelica.asc -O- | sudo apt-key add -`

**Comment by sjoelund** [comment]
The build dependencies of OpenModelica should not contain the testsuite dependencies. We should keep a separate list for that. You could link to the Dockerfile for the image Jenkins uses and install those dependencies: https://github.com/OpenModelica/OpenModelicaBuildScripts/blob/master/docker/Dockerfile.build-deps

**Comment by AnHeuermann** [comment]
Do you meen this line?
```
apt-get install -qy gnupg wget ca-certificates apt-transport-https devscripts equivs python3-pip python-pip libmldbm-perl docker.io sudo git subversion texlive-base texlive-latex-extra latexmk gnuplot-nox doxygen poppler-utils flex libgfortran3 aspell bibtex2html zip unzip ocl-icd-opencl-dev cpio xsltproc inkscape g++-4.8 g++-5 g++-6 g++-7 texlive-lang-greek
```

**Comment by sjoelund** [comment]
You need all of the lines if you want to test OpenCL, build the documentation. The apt-get line installs the bare minimum for the testsuite (and some more like extra g++ versions)

**Comment by JKRT** [comment]
> 
> 
> And I have still some failing tests with optimization and measuring time:
> 
> ```
> ./openmodelica/cruntime/optimization/basic/TFC6.mos
> ./openmodelica/cruntime/optimization/basic/testAlgLoop6.mos
> ./openmodelica/cruntime/optimization/basic/testAlgLoop7.mos
> ./openmodelica/cruntime/optimization/basic/testDerInput.mos
> ./openmodelica/cruntime/optimization/benchmark/runDrumBoiler.mos
> ./simulation/modelica/built_in_functions/MeasureTime.mos
> ```
> 
> @JKRT Did you run the test suite on your WSL and got a better result? Probably I'm missing one or two programs still.

Ah Sorry. No, I did not, got caught up into something else! Basically, the only tests I run are usually the ones for MetaModelica. When running the entire thing I do get some errors since the values on my machine seem to slightly of sometimes

**Comment by sjoelund** [comment]
I'm not sure it's good to have the dependencies listed in the README due to them changing. Might be better to link the dockerfile for every dependency.

**Comment by AnHeuermann** [comment]
I updated the dependencies. For absolute beginners there is a "minimal" version to get the testsuite running and for everybody else the Dockerfile is linked as you suggested.

**Comment by AnHeuermann** [comment]
@sjoelund Can the PR be merged?

**Pull request merged**